### PR TITLE
Feat: Creates Picture Component

### PR DESCRIPTION
**Pull request by eduardoformiga**
## What's the purpose of this pull request?

ported from https://github.com/vtex-sites/base.store/pull/489

Creates the `Picture` component and `useSources` hook.

## How does it work?

While analyzing some of our client's projects, I noticed a problem regarding the way the images were loading when we have multiple breakpoints, like mobile and desktop.

Taking into consideration that we may want to load different images according to the device (Art Direction) and we were controlling what will be shown using CSS (`display-mobile` and `hidden-mobile` classes), the browser was downloading at least two images, one time for mobile and another for desktop.

Example:

```js
<Banner>
  <Image
    className="display-mobile"
    alt={altMobile}
    src={srcMobile}
    width={widthMobile}
    height={heightMobile}
  />
  <Image
    className="hidden-mobile"
    alt={altDesktop}
    src={srcDesktop}
    width={widthDesktop}
    height={heightDesktop}
  />
</Banner>
```

Since the HTML5 provides the `<picture>` tag along with `<source>`, `srcset`, and `media` queries, we can pass some of them and the browser will download just the best option for the device screen width and pixel density. So, we can pass these options instead of controlling everything using CSS, what decreases the DOM size, CSS classes, and also we can save the bandwidth downloading smaller images. Thus, speeds up page load times.

`<picture>` will download all image sources as needed and dynamically show the appropriate at the time, even if the user changes the dimensions of the page. Its main use is Art Direction (when you want to show a differently shaped picture under different conditions. Maybe you want to show a cropped version of the picture on mobile or scale some text (e.g. small text that would become unreadable when scaled).

Why not use the `Image` component?

The `Image` component is recommended when we just have one image (mobile or desktop) and we would like to have **multiple copies of the same image**, preserving the aspect ratio for all image copies with more or fewer pixels. When we would like to have **different images for different screens**, the `<picture>` tag is recommended.

Multiples copies of the same image:
164313193-075ccc42-ac27-4e6a-9e45-8dbe5448b7f3

Different images for different screens:
164313237-5e0d8745-f2f4-4d02-8770-061cb364c46f
164313482-d788ee45-d04d-47d6-9b36-66807d01eba8

## How to test it?

So far we don't have the Art Direction problem (maybe on Hero or Tiles afterward?). If we decided to change the implementation of these components we can test better.

Example of use:

```js
<Picture
  sources={[
    {
      src: srcMobile,
      width: widthMobile,
      height: heightMobile,
      media: '(max-width: 767px)',
    },
    {
      src: srcDesktop,
      width: widthDesktop,
      height: heightdesktop,
      media: '(min-width: 768px)',
    },
  ]}
  img={{
    src: srcFallback,
    width: widthFallback,
    height: heightFallback,
    alt: altFallback,
    loading: "lazy",
  }}
/>
```

Final markup example:

```js
<picture>
  <source srcset="https://assets.vtex.app/328x435/image328.png 328w, https://assets.vtex.app/656x870/image656.png 656w, https://assets.vtex.app/984x1305/image984.png 984w" media="(max-width: 767px)">
  <source srcset="https://assets.vtex.app/1304x636/image1304.png 1304w, https://assets.vtex.app/2608x1272/image2608.png 2608w, https://assets.vtex.app/3912x1908/image3912.png 3912w" media="(min-width: 768px)">
  <img src="https://store.vtexassets.com/assets/image.png" width="1304" height="636" alt="Alternative text" loading="lazy">
</picture>
```

## References

https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_image
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/picture
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/source
https://blog.bitsrc.io/why-you-should-use-picture-tag-instead-of-img-tag-b9841e86bf8b
https://www.w3schools.com/html/html_images_picture.asp
https://web.dev/preload-responsive-images/#preload-and-lesspicturegreater
https://github.com/vtex-sites/base.store/pull/401

## Checklist

<em>You may erase this after checking them all ;)</em>

- [x] Added an entry in the `CHANGELOG.md` at the beginning of its due section. [The latest version should come first](https://keepachangelog.com/en/1.0.0/#:~:text=The%20latest%20version%20comes%20first.).
- [x] Added the PR number with the PR link at the entry in the `CHANGELOG.md`. E.g., *New items in the `pull_request_template.md` ([#12](https://github.com/vtex-sites/gatsby.store/pull/12))* 

- PR description
- [x] Updated the Storybook - *if applicable*.
- [x] Added a label according to the PR goal - `Breaking change`, `Enhancement`, `Bug` or `Chore`.
- [x] Added the component, hook, or pathname in-between backticks (``) *- If applicable*. E.g., *`ComponentName` component*.
- [x] Identified the function or parameter in the PR *- If applicable*. E.g., *`useWindowDimensions` hook*.
- [x] For documentation changes, ping @carolinamenezes or @Mariana-Caetano to review and update the changes.

### CODE REVIEW

File: src/components/ui/Image/Picture.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  preload = false,
+  ...otherProps
+}: Props) {
+  const sources = useSources(sourceOptions)
+
+  return (
+    <picture>
+      {sources.map(({ srcSet, media, height, preloadLinks }, index) => {
+        return (
+          <Fragment key={index}>
```
Comment (saranicoly) [inline src/components/ui/Image/Picture.tsx:27]: I'm wondering if using the index as a key (here and in the line 41) won't cause any issues (as exemplified [here](https://robinpokorny.com/blog/index-as-a-key-is-an-anti-pattern/)).

Comment (eduardoformiga) [reply src/components/ui/Image/Picture.tsx:27]: Well noticed, Sara.

I think this picture is the [Exception from the rule](https://robinpokorny.com/blog/index-as-a-key-is-an-anti-pattern/#:~:text=Exception%20from%20the%20rule), though. Do you notice any case where the `sources` can change, reordered, or filtered?

**Comment by hellofanny** [comment]
@eduardoformiga Do you think we can close this PR for now? Since we already closed it on [nextjs](https://github.com/vtex-sites/nextjs.store/pull/52) repo?

**Comment by eduardoformiga** [comment]
> @eduardoformiga Do you think we can close this PR for now? Since we already closed it on [nextjs](https://github.com/vtex-sites/nextjs.store/pull/52) repo?

Yeah, let's close it.

**Pull request closed without merging**